### PR TITLE
scripts/test-sim-backwards: usability improvements

### PR DESCRIPTION
**Pull request by bryanchriswhite**
What: 
  + Add `git worktree remove ...` to `cleanup` function.
  + Add comit-ish to `git worktree add ...` for `$BRANCH_DIR`

Why: In a non-ephemeral environment, the script would run once and then error on subsequent runs until the git worktree was cleaned up. The head of the new worktrees also were not guaranteed to be the head of the current branch.

Please describe the tests: N/A
 
Please describe the performance impact: N/A

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: scripts/test-sim-backwards.sh [hunk 1]
```diff
@@ -31,13 +33,14 @@ make -C "$RELEASE_DIR" install-sim
 STORJ_NETWORK_HOST4=${STORJ_NETWORK_HOST4:-127.0.0.1}
 STORJ_SIM_POSTGRES=${STORJ_SIM_POSTGRES:-""}
 
-# setup the network
 if [ -z ${STORJ_SIM_POSTGRES} ]; then
-    storj-sim -x --host $STORJ_NETWORK_HOST4 network setup
-else
-    storj-sim -x --host $STORJ_NETWORK_HOST4 network --postgres=$STORJ_SIM_POSTGRES setup
+    echo "Postgres is required for the satllite DB. Exiting."
```
Comment (egonelbre) [inline scripts/test-sim-backwards.sh]: typo

Comment (bryanchriswhite) [reply scripts/test-sim-backwards.sh]: dang

File: scripts/test-sim-backwards.sh [hunk 2]
```diff
@@ -6,6 +6,8 @@ TMP=$(mktemp -d -t tmp.XXXXXXXXXX)
 export STORJ_NETWORK_DIR=$TMP
 cleanup(){
     rm -rf "$STORJ_NETWORK_DIR"
+    git worktree remove -f "$RELEASE_DIR"
+    git worktree remove -f "$BRANCH_DIR"
```
Comment (JessicaGreben) [inline scripts/test-sim-backwards.sh]: does this work for you, i kept getting errors trying to add worktree remove?

File: scripts/test-sim-backwards.sh [hunk 3]
```diff
@@ -31,13 +33,14 @@ make -C "$RELEASE_DIR" install-sim
 STORJ_NETWORK_HOST4=${STORJ_NETWORK_HOST4:-127.0.0.1}
 STORJ_SIM_POSTGRES=${STORJ_SIM_POSTGRES:-""}
 
-# setup the network
 if [ -z ${STORJ_SIM_POSTGRES} ]; then
-    storj-sim -x --host $STORJ_NETWORK_HOST4 network setup
-else
-    storj-sim -x --host $STORJ_NETWORK_HOST4 network --postgres=$STORJ_SIM_POSTGRES setup
+    echo "Postgres is required for the satellite DB. Exiting."
+    exit 1
```
Comment (JessicaGreben) [inline scripts/test-sim-backwards.sh]: good idea!

Comment (bryanchriswhite) [reply scripts/test-sim-backwards.sh]: thank @egonelbre :p

File: scripts/test-sim-backwards.sh (hunk 2 above)
Comment (bryanchriswhite) [reply scripts/test-sim-backwards.sh]: interesting, this does work for me i'm using `git version 2.22.0` fwiw

Comment (bryanchriswhite) [reply scripts/test-sim-backwards.sh]: what does the error look like?

Comment (JessicaGreben) [reply scripts/test-sim-backwards.sh]: at the end when cleanup happens i get an error like this 

```fatal: '/var/folders/s_/n4czty1x4tj297495m66bcsc0000gp/T/tmp.XXXXXXXXXX.Ybi0nSkW/release' is not a working tree```

Comment (JessicaGreben) [reply scripts/test-sim-backwards.sh]: moving `rm -rf "$STORJ_NETWORK_DIR"` after the worktree remove, fixed it

**Pull request merged**